sdks/python/apache_beam/examples/complete/distribopt.py

```diff
@@ -314,7 +314,7 @@ def format_output(element):
   return result
 
 
-def run(argv=None):
+def run(argv=None, save_main_session=True):
```

Comment:
Can we determine this instead of adding a parameter? Since this change affects examples, I am worried that users will transfer this pattern into their pipelines (or will have to reason whether they should), and it might be unnecessary complication.

Perhaps we could:
```
# Explain why we do this, where it's not explained
pipeline_options.view_as(SetupOptions).save_main_session = __name__ == '__main__'
```
WDYT?

Reply:
I think the best solution would be to remove save_main_session from examples altogether, even if that means splitting them into 2 modules. Since that is out of scope for the item I'm working on, I opted for the solution I proposed. I believe my solution is necessary and simpler than checking the magic variable `__name__`.

Reply:
I am not sure adding another parameter is simpler, for example it may create a question to users such as:

_Why do some parameters to `run()` are passed via parsing argv, and some via additional parameter, how do we chose which method to use? Oh wait, `save_main_session` is actually defined as an argument in `SetupOptions`, why the example does not reuse that option?_ 

It is not obvious to users that we need this parameter just to make testing simpler. If you think that adding an extra argument is easier, consider naming it something like `testing=False` to indicate that this codepath is required for Beam integration tests. 

In my opinion the less parameters users have to reason about, the easier the example, and I still think this parameter is not necessary, but perhaps the following would be more reader-friendly:
```
if __name__ == '__main__':
   pipeline_options.view_as(SetupOptions).save_main_session = True
```
Feel free to disagree or solicit another opinion. If you still think your solution is better, consider this review LGTM.